src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -584,10 +569,12 @@ private struct CompilationInfo
                 public Compilation Compilation { get; }
                 public bool HasCompleteReferences { get; }
 
-                public CompilationInfo(Compilation compilation, bool hasCompleteReferences)
+                public CompilationInfo(Compilation compilation, ProjectInfo projectInfo, bool hasCompleteReferences)
                 {
                     this.Compilation = compilation;
-                    this.HasCompleteReferences = hasCompleteReferences;
+
+                    // to have complete reference, projectInfo should be complete and we could create complete reference for this compilation.
+                    this.HasCompleteReferences = projectInfo.IsComplete && hasCompleteReferences;
```

Comment:
My complaint is we're strangely conflating two things. If the project was missing source, but not references, we have no way to say it's missing just source references. So HasCompleteReferences should still be true, but it'd be false. I don't care which way this goes; I'm OK with us dropping "references" from "completeness" or adding it everywhere, but let's be consistent with it.